src/handlers/InputAdded.ts

```diff
@@ -33,40 +33,45 @@ export default class InputAdded implements Handler {
         private inputStorage: Map<String, Input>,
     ) {}
 
-    async handlePayload(
+    private async prepareErc20Deposit(
         input: Input,
         block: BlockData,
         ctx: DataHandlerContext<Store>,
+        opts: {
+            inputId: string;
+        },
     ) {
-        if (input.msgSender == ERC20PortalAddress) {
-            const success = getNumber(dataSlice(input.payload, 0, 1)) == 1; // 1 byte for boolean (not used?)
-            const tokenAddress = dataSlice(input.payload, 1, 21).toLowerCase(); // 20 bytes for address
-            const from = dataSlice(input.payload, 21, 41).toLowerCase(); // 20 bytes for address
-            const amount = getUint(dataSlice(input.payload, 41, 73)); // 32 bytes for uint256
-
-            let token = this.tokenStorage.get(tokenAddress) as Token;
-            if (!token) {
-                const contract = new ERC20(ctx, block.header, tokenAddress);
-                const name = await contract.name();
-                const symbol = await contract.symbol();
-                const decimals = await contract.decimals();
-                token = new Token({ id: tokenAddress, name, symbol, decimals });
-                this.tokenStorage.set(tokenAddress, token);
-                ctx.log.info(`${tokenAddress} (Token) stored`);
-            }
-            const deposit = new Erc20Deposit({
-                id: input.id,
-                amount,
-                from,
-                token,
-            });
-            return deposit;
+        if (input.msgSender !== ERC20PortalAddress) return undefined;
+
+        const success = getNumber(dataSlice(input.payload, 0, 1)) == 1; // 1 byte for boolean (not used?)
```

Comment:
Hi @brunomenezes this variable is not used. Maybe consider remove it?

Reply:
Hello @dandheedge, that is addressed in the PR #22. I need the #22 to be reviewed first since the targeted branch is this one `refactor/erc-20-deposit-handling`